instance/models/openedx_instance.py

```diff
@@ -117,12 +117,21 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         for appserver in active_appservers:
             server_name = "appserver-{}".format(appserver.pk)
             ip_address = appserver.server.public_ip
+            if not ip_address:
+                appserver.logger.error(
+                    "Active appserver does not have a public IP address. This should not happen."
+                    "Calling update_status()"
+                )
+                appserver.server.update_status()
+                ip_address = appserver.server.public_ip
             if ip_address:
                 appserver_vars.append(dict(ip_address=ip_address, name=server_name))
             else:
-                appserver.logger.error(
-                    "Active appserver does not have a public IP address. This should not happen."
+                self.logger.error(
+                    "Active appserver still does not have a public IP address after calling update_status()."
+                    "Deconfiguring the load balancer backend."
                 )
+                return [], []
```

Comment:
This does nothing different than what it already would've, actually, for the case we're trying to prevent. Notice that `appserver_vars` will be empty if we don't return here if there is an issue retrieving the public IP, and it'll just go down to line 136, and do the same thing.

As mentioned earlier, we want to kill the call stack, because this case "should not happen" and if it does, it's too dangerous to proceed and potentially accidentally deconfigure an active and healthy backend.